packages/firestore/src/lite-api/database.ts

```diff
@@ -132,12 +134,20 @@ export class Firestore implements FirestoreService {
   }
 
   _delete(): Promise<void> {
-    if (!this._terminateTask) {
+    if (this._terminateTask === 'notTerminated') {
       this._terminateTask = this._terminate();
     }
     return this._terminateTask;
   }
 
+  async _restart(): Promise<void> {
+    if (this._terminateTask === 'notTerminated') {
```

Comment:
Add some comment might help further. Something along the line of "keeping it as 'notTerminated' such that...".